components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBSaver.java

```diff
@@ -43,45 +59,122 @@ public void init() throws ArtifactSynchronizerException {
+                        "code : " + httpResponse.getStatusLine().getStatusCode() );
+                throw new ArtifactSynchronizerException("Error while Saving Artifacts to DB");
             }
-            apiMgtDAO.addGatewayPublishedAPIArtifacts(apiId, gatewayLabel,
-                    byteArrayInputStream, gatewayRuntimeArtifactsAsBytes.length, gatewayInstruction, dbQuery);
+        } catch (MalformedURLException e) {
+            String msg = "Error while constructing Synapse POST URL";
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
+        } catch (IOException e) {
+            String msg = "Error while executing the http client";
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
+        }
+    }
 
-            if (log.isDebugEnabled()) {
-                log.debug("Successfully saved Artifacts of " + apiName);
-            }
-        } catch (APIManagementException e) {
-            throw new ArtifactSynchronizerException("Error saving Artifacts to the DB", e);
+    private HttpResponse invokeService(String endpoint, JSONObject revokeRequestPayload) throws
+            IOException {
+        URL synapsePostURL = new URL(endpoint);
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance()
+                .getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String username = config.getFirstProperty(APIConstants.API_KEY_VALIDATOR_USERNAME);
+        String password = config.getFirstProperty(APIConstants.API_KEY_VALIDATOR_PASSWORD);
+        byte[] credentials = Base64.encodeBase64((username + ":" + password).getBytes
+                (APIConstants.DigestAuthConstants.CHARSET));
+        int synapsePort = synapsePostURL.getPort();
+        String synapseProtocol = synapsePostURL.getProtocol();
+        HttpClient httpClient = APIUtil.getHttpClient(synapsePort, synapseProtocol);
+
+        if (revokeRequestPayload != null) {
+             HttpPost method = new HttpPost(endpoint);
+             method.setHeader("Authorization", "Basic " + new String(credentials,
+                    APIConstants.DigestAuthConstants.CHARSET));
+             StringEntity requestEntity = new StringEntity(revokeRequestPayload.toString());
+             requestEntity.setContentType(APIConstants.APPLICATION_JSON_MEDIA_TYPE);
+             method.setEntity(requestEntity);
+             return executeHTTPRequest(method, httpClient);
+
+        } else {
+             HttpGet method = new HttpGet(endpoint);
+             method.setHeader("Authorization", "Basic " + new String(credentials,
+                    APIConstants.DigestAuthConstants.CHARSET));
+             return executeHTTPRequest(method, httpClient);
         }
+    }
 
+    private HttpResponse executeHTTPRequest(HttpRequestBase method, HttpClient httpClient ) throws IOException {
```

Comment:
Can't we add this method to APIUtil and use it by both DBSaver and DBretriever?

Reply:
https://github.com/wso2/carbon-apimgt/pull/8828/commits/0007abb08c126e094b757b72e03eba846516e404